src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -935,75 +841,62 @@ void CastSpellInfoHelpers::CastSpell() {
-                    dist_Y = abs(pActors[monster_id].vPosition.y -
-                                 pParty->vPosition.y);
-                    dist_Z = abs(pActors[monster_id].vPosition.z -
-                                 pParty->vPosition.z);
+                    dist_X = abs(pActors[monster_id].vPosition.x - pParty->vPosition.x);
+                    dist_Y = abs(pActors[monster_id].vPosition.y - pParty->vPosition.y);
+                    dist_Z = abs(pActors[monster_id].vPosition.z - pParty->vPosition.z);
                     int count = int_get_vector_length(dist_X, dist_Y, dist_Z);
                     if ((double)count <= 307.2) {
-                        v701.x = 0;
-                        v701.y = 0;
-                        v701.z = 0;
-                        pSpellSprite.containing_item.Reset();
-                        pSpellSprite.spell_id = pCastSpell->uSpellID;
-                        pSpellSprite.spell_level = spell_level;
-                        pSpellSprite.spell_skill = spell_mastery;
-                        pSpellSprite.uObjectDescID = pObjectList->ObjectIDByItemID(pSpellSprite.uType);
-                        pSpellSprite.uAttributes = 0;
+                        InitSpellSprite( &pSpellSprite, spell_level, spell_mastery, pCastSpell);
                         pSpellSprite.uSectorID = 0;
-                        pSpellSprite.uSpriteFrameID = 0;
                         pSpellSprite.field_60_distance_related_prolly_lod = 0;
-                        pSpellSprite.spell_caster_pid = PID(OBJECT_Player, pCastSpell->uPlayerID);
                         pSpellSprite.uFacing = 0;
-                        pSpellSprite.uSoundID = (short)pCastSpell->sound_id;
                         pSpellSprite.vPosition.x = pActors[monster_id].vPosition.x;
                         pSpellSprite.vPosition.y = pActors[monster_id].vPosition.y;
                         pSpellSprite.vPosition.z = pActors[monster_id].vPosition.z - (int)((double)pActors[monster_id].uActorHeight * -0.8);
                         pSpellSprite.spell_target_pid = PID(OBJECT_Actor, spell_targeted_at);
-                        Actor::DamageMonsterFromParty(PID(OBJECT_Item, pSpellSprite.Create(0, 0, 0, 0)), monster_id, &v701);
-                        spell_sound_flag = true;
+                        Actor::DamageMonsterFromParty(PID(OBJECT_Item, pSpellSprite.Create(0, 0, 0, 0)), monster_id, &spell_velocity);
                     } else {
-                        GameUI_SetStatusBar(LSTR_SPELL_FAILED);
-                        pAudioPlayer->PlaySound(SOUND_spellfail0201, 0, 0, -1, 0, 0);
-                        pCastSpell->uSpellID = 0;
+                        SpellFailed(pCastSpell, LSTR_SPELL_FAILED);
+                        pPlayer->SpendMana(uRequiredMana); // decrease mana on failure
+                        continue;
                     }
+                } else {
+                    // Previous behavoiur - decrease mana but do not play sound if
```

Comment:
Add a `TODO: revisit, maybe we want to properly play sound and display a message here?`